discord/ext/commands/converter.py

```diff
@@ -524,6 +525,33 @@ async def convert(self, ctx, argument):
         except Exception as exc:
             raise BadInviteArgument() from exc
 
+class GuildConverter(IDConverter):
+    """Converts to a :class:`~discord.Guild`.
+
+    The lookup strategy is as follows (in order):
+
+    1. Lookup by ID.
+    2. Lookup by name.
+
+    .. versionadded:: 1.7
+    """
+
+    async def convert(self, ctx, argument):
+        match = self._get_id_match(argument)
+        result = None
+
+        if match is not None:
+            guild_id = int(match.group(1))
+            result = ctx.bot.get_guild(guild_id)
+            return result
```

Comment:
This could return None instead of raising GuildNotFound

Reply:
That wouldn't match the behaviour of any of the current converters

Reply:
That's what I mean? I didn't change anything and it currently can return None for an wrong id